app/src/main/java/com/example/topmovies/repository/MovieRepository.kt

```diff
@@ -1,58 +1,99 @@
 package com.example.topmovies.repository
 
-import com.example.topmovies.model.MovieDetails
-import com.example.topmovies.model.MovieObject
+import com.example.topmovies.database.dao.MovieDetailsDao
+import com.example.topmovies.database.dao.MoviesDao
+import com.example.topmovies.model.*
 import com.example.topmovies.retrofit.MoviesApi
 import retrofit2.Call
 import retrofit2.Callback
 import retrofit2.Response
+import java.util.concurrent.ExecutorService
+
+class MovieRepository constructor(
```

Comment:
Please regroup the methods a bit. keep the close to each other based on their role, i.e. insert and update shouldn't be apart.